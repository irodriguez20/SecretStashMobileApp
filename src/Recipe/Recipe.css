.Recipe {
  --recipe-height: 88px;
  background-color: #8a6f78;
  color: #181417;
  height: var(--note-height);
  margin-bottom: 8px;
  max-width: 380px;
  padding: 8px 0 8px 8px;
  border-radius: 3px;
  position: relative;
}

.Recipe__delete {
  --recipe-button-height: 32px;
  background-color: rgb(27, 27, 61);
  border: 0;
  border-radius: 3px;
  color: #f0ebeb;
  float: right;
  font-size: 12px;
  height: var(--recipe-button-height);
  position: static;
  top: calc((var(--recipe-button-height) + 32px) * -1);
  /* transform-origin: right bottom; */
  /* transform: rotate(-90deg); */
  width: calc(var(--recipe-height));
}
.Recipe__delete:hover {
  background-color: #c8556a;
  color: white;
  cursor: pointer;
  font-weight: bold;
}

.Recipe__edit {
  --recipe-button-height: 32px;
  background-color: rgb(27, 27, 61);
  border: 0;
  border-radius: 3px;
  color: #f0ebeb;
  width: 50px;
  /* float: left; */
  font-size: 12px;
  height: var(--recipe-button-height);
  /* position: static; */
  top: calc((var(--recipe-button-height) + 32px) * -1);
}

.Recipe__edit:hover {
  background-color: #c8556a;
  color: white;
  cursor: pointer;
  font-weight: bold;
}

.Recipe__title {
  font-family: var(--custom-font);
  font-size: 20px;
  margin: 0;
  max-width: 80%;
  min-height: 24px;
}
.Recipe__title a {
  color: #ffffff;
  display: block;
  text-decoration: none;
}
.Recipe__title a:hover {
  color: #f5e553;
}
