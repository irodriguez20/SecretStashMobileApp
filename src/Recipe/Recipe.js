import React, { Component } from 'react';
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import Context from '../Context'
import PropTypes from "prop-types";
import { Route, Link } from 'react-router-dom';

class Recipe extends Component {
    static propTypes = {
        history: PropTypes.shape({
            push: () => { }
        }).isRequired,
    };

    static contextType = Context;

    handleClickDelete = e => {
        e.preventDefault();
        const recipeId = this.props.id;
    }

    render() {

        const { name, id } = this.props;
        return (
            <div className="Recipe">
                <main>
                    <header className="banner">
                        <h2><Link to={`/recipe/${id}`}>{name}</Link></h2>
                    </header>
                    <Link to={`/edit/${id}`}>
                        <button type="edit">Edit</button></Link>

                    <button onClick={this.handleClickDelete} type="button">
                        <FontAwesomeIcon icon="trash-alt" />
                        {" "}
                        Delete
                    </button>
                </main>
            </div>
        );
    }
}

export default Recipe;