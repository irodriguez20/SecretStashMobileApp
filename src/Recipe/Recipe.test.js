import React from "react";
import Recipe from "./Recipe";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";

describe(`Recipe component `, () => {
    const props = {
        id: "a",
        name: "test-class-name",
        timetomake: "test-time-to-make",
        description: "test description",
        ingredients: "test ingredients",
        steps: "test steps",
    };

    it("renders a .Recipe by default", () => {
        const wrapper = shallow(<Recipe />);
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it("renders the Recipe given props", () => {
        const wrapper = shallow(<Recipe {...props} />);
        expect(toJson(wrapper)).toMatchSnapshot();
    });
});
