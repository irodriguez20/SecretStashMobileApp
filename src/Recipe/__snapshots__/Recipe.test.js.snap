// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Recipe component  renders a .Recipe by default 1`] = `
<div
  className="Recipe"
>
  <section>
    <header
      className="Recipe__title"
    >
      <h2>
        <Link
          to="/recipe/undefined"
        />
      </h2>
      <h3 />
    </header>
    <blockquote />
    <Link
      to="/edit/undefined"
    >
      <button
        type="edit"
      >
        Edit
      </button>
    </Link>
    <button
      className="Recipe__delete"
      onClick={[Function]}
      type="button"
    >
      <FontAwesomeIcon
        border={false}
        className=""
        fixedWidth={false}
        flip={null}
        icon="trash-alt"
        inverse={false}
        listItem={false}
        mask={null}
        pull={null}
        pulse={false}
        rotation={null}
        size={null}
        spin={false}
        swapOpacity={false}
        symbol={false}
        title=""
        transform={null}
      />
       
      Delete
    </button>
  </section>
</div>
`;

exports[`Recipe component  renders the Recipe given props 1`] = `
<div
  className="Recipe"
>
  <section>
    <header
      className="Recipe__title"
    >
      <h2>
        <Link
          to="/recipe/a"
        >
          test-class-name
        </Link>
      </h2>
      <h3>
        test-time-to-make
      </h3>
    </header>
    <blockquote>
      test description
    </blockquote>
    <Link
      to="/edit/a"
    >
      <button
        type="edit"
      >
        Edit
      </button>
    </Link>
    <button
      className="Recipe__delete"
      onClick={[Function]}
      type="button"
    >
      <FontAwesomeIcon
        border={false}
        className=""
        fixedWidth={false}
        flip={null}
        icon="trash-alt"
        inverse={false}
        listItem={false}
        mask={null}
        pull={null}
        pulse={false}
        rotation={null}
        size={null}
        spin={false}
        swapOpacity={false}
        symbol={false}
        title=""
        transform={null}
      />
       
      Delete
    </button>
  </section>
</div>
`;
